### PR TITLE
chore(deps): update sqlparser requirement from 0.44.0 to 0.45.0

### PR DESCRIPTION
**Pull request by Jefffrey**
## Which issue does this PR close?

Supersedes #10085

## What changes are included in this PR?

Handle some sqlparser API changes. Have tried to keep changes minimum.

### CODE REVIEW

File: datafusion/common/src/functional_dependencies.rs [hunk 1]
```diff
@@ -68,11 +68,26 @@ impl Constraints {
...
+                    // Get primary key and/or unique indices in the schema:
+                    let indices = columns
+                        .iter()
+                        .map(|pk| {
+                            let idx = field_names
+                                .iter()
+                                .position(|item| *item == pk.value)
+                                .ok_or_else(|| {
+                                    DataFusionError::Execution(
+                                        "Unique key doesn't exist".to_string(),
```
Comment (andygrove) [inline datafusion/common/src/functional_dependencies.rs:82]: Can we include the field name in the error message to make it easier for the user to debug when they hit this error?

Comment (comphead) [reply datafusion/common/src/functional_dependencies.rs:82]: as well as unique key details

Comment (Jefffrey) [reply datafusion/common/src/functional_dependencies.rs]: Sure thing, done :+1:

**Review by alamb — APPROVED** [review]
Thank you very much @Jefffrey  -- looks great to me 🙏

File: datafusion/common/src/functional_dependencies.rs [hunk 2]
```diff
@@ -68,32 +68,48 @@ impl Constraints {
...
-                                .position(|item| *item == pk.value)
+                                .position(|item| *item == u.value)
                                 .ok_or_else(|| {
+                                    let name = name
+                                        .as_ref()
+                                        .map(|name| format!("with name '{name}' "))
+                                        .unwrap_or("".to_string());
                                     DataFusionError::Execution(
-                                        "Primary key doesn't exist".to_string(),
+                                        format!("Column for unique constraint {}not found in schema: {}", name,u.value)
```
Comment (alamb) [inline datafusion/common/src/functional_dependencies.rs:86]:
```suggestion
                                        format!("Column for unique constraint {} not found in schema: {}", name,u.value)
```

**Review by alamb — APPROVED** [review]

File: datafusion/common/src/functional_dependencies.rs (hunk 2 above)
Comment (Jefffrey) [reply datafusion/common/src/functional_dependencies.rs:86]: Ah the missing whitespace is actually intentional here as the unique constraint may be unnamed :slightly_smiling_face:

Comment (alamb) [reply datafusion/common/src/functional_dependencies.rs:86]: Got it -- sorry -- maybe we can add a test or something of the error message in a follow on PR to make it clearer.

**Comment by alamb** [comment]
Thanks @Jefffrey !

**Pull request merged**